src/coreclr/jit/importervectorization.cpp

```diff
@@ -447,7 +452,7 @@ GenTree* Compiler::impExpandHalfConstEqualsSWAR(
             return nullptr;
         }
 
-        secondIndir = gtNewCastNode(TYP_LONG, secondIndir, true, TYP_LONG);
+        secondIndir = gtNewCastNode(TYP_LONG, secondIndir, true, TYP_ULONG);
```

Comment:
```suggestion
        secondIndir = gtNewCastNode(TYP_LONG, secondIndir, true, TYP_LONG);
```
(These two are the same)

Reply:
ULONG looks clearer to me in terms of intention, but I assume it's better to keep it canonized

Reply:
Yeah, well, doesn't really matter. One minor benefit is keeping the diff small.